core/kernel.go

```diff
@@ -202,6 +205,44 @@ func (kern *Kernel) LoadDump(genesisDoc *genesis.GenesisDoc, restoreFile string,
 	return nil
 }
 
+// DialPeersFromGenesis allows us to autoconnect local validators on start
+func (kern *Kernel) DialPeersFromGenesis(nodes []string) {
+	pb := make(map[string]string, len(nodes))
+	for _, entry := range nodes {
+		target := strings.Split(entry, "=")
+		if len(target) != 2 {
+			kern.Logger.InfoMsg(fmt.Sprintf("Entry '%s' is not valid", entry))
+		}
+		pb[target[0]] = target[1]
+	}
+
+	validators := kern.Blockchain.GenesisDoc().Validators
+	for _, val := range validators {
+		if _, ok := pb[val.Name]; !ok {
+			continue
+		}
+		host, port, err := net.SplitHostPort(pb[val.Name])
```

Comment:
We have called this value `moniker` which has a specific meaning to tendermint. The `Name` in the genesis validator struct has no fixed relation to the validator moniker which a tendermint node can choose for itself. Therefore we should not be calling this `moniker` since implies that somehow this is the same as `Name`